anchore_engine/analyzers/__init__.py

```diff
@@ -0,0 +1,113 @@
+import os
+import re
+import time
+import collections
+
+import anchore_engine.utils
+from anchore_engine.subsys import logger
+from pkg_resources import resource_filename
+from . import syft
+from . import binary
+from . import utils
+
+
+def run(configdir, imageId, unpackdir, outputdir, copydir):
+    analyzer_report = collections.defaultdict(dict)
+    _run_analyzer_modules(analyzer_report, configdir, imageId, unpackdir, outputdir)
+    _run_syft(analyzer_report, copydir)
+    _run_internal_analyzers(analyzer_report, unpackdir)
+
+    apply_hints(analyzer_report, unpackdir)
+
+    return analyzer_report
+
+
+def apply_hints(analyzer_report, unpackdir):
+    # apply content hints, overriding values that are there
+    # note: upstream of this processing overwrites the hints file location if
+    # the config does not explicitly enable hints processing, effectively disabling
+    # hints processing.
+    for engine_entry in utils.content_hints(unpackdir=unpackdir):
+        pkg_type = engine_entry.get("type")
+        if pkg_type:
+            handler = syft.modules_by_engine_type[pkg_type]
+            handler.save_entry(analyzer_report, engine_entry)
+
+
+def _run_analyzer_modules(analyzer_report, configdir, imageId, unpackdir, outputdir):
+    anchore_module_root = resource_filename("anchore_engine", "analyzers")
+    analyzer_root = os.path.join(anchore_module_root, "modules")
+    for f in list_modules():
+        cmdstr = " ".join([f, configdir, imageId, unpackdir, outputdir, unpackdir])
+        with anchore_engine.utils.timer(
+            "Executing analyzer %s".format(repr(f)), log_level="debug"
+        ):
+            try:
+                rc, sout, serr = anchore_engine.utils.run_command(cmdstr)
+                sout = anchore_engine.utils.ensure_str(sout)
+                serr = anchore_engine.utils.ensure_str(serr)
+                if rc != 0:
+                    logger.error(
+                        "command failed: cmd=%s exitcode=%s stdout=%s stderr=%s".format(
+                            repr(cmdstr),
+                            repr(rc),
+                            repr(sout.strip()),
+                            repr(serr.strip()),
+                        )
+                    )
+                else:
+                    logger.debug(
+                        "command succeeded: cmd=%s stdout=%s stderr=%s".format(
+                            repr(cmdstr), repr(sout.strip()), repr(serr.strip())
+                        )
+                    )
+            except Exception as err:
```

Comment:
there is no need to capture `err` here

Reply:
Just following along and learning Python 😄 — would we not want to include the error in the log message?

Reply:
@luhring the `logger.exception` takes care of that for you